### ISSUE TITLE
Mapview.center() is often working late or less accurate

### ISSUE THREAD

**Issue by johan974** [osmdroid/osmdroid#987]
## Issue Type

[X] Bug

## Description and/or steps/code to reproduce the problem
I am using Osmdroid for over 4 years weekly. I like using the new osmdroid very much. Thank you!
Since using the osmdroid 6.0.1 centering on the mapview is often done very late or less accurate.  This is a bit annoying. 
No code has been changed on this part of the App. 

## Environment
Android 7.0.1. Osmdroid 6.0.1. 

### If it's a bug, version(s) of android this affects:
Mapview.center( ... ) 

### Version of osmdroid the issue relates to:
6.0.1

**Comment by monsieurtanuki** [comment]
> centering on the mapview is often done very late or less accurate

@johan974 There were indeed changes in 6.0.1 around map centering.
Could you please be more specific about the inaccuracy you experience, like how to reproduce it (zoom level, expected center, actual center), and about the "very late" part of your comments?
Please note that `MapView.setMapCenter` is now deprecated, and that `MapView.setExpectedCenter` should be used instead.

**Comment by johan974** [comment]
Every say 1 second I get a new GPS update. Using that I center the map, draw a line from my current location to the target. I see that especially after awakening my mobile (during navigation) it takes 5 seconds to get a good center. 
Zoom level - often near 18 - 20. 
The map center is set via: mapView.getController().setCenter( gp);

**Comment by monsieurtanuki** [comment]
Then I would say it's a problem of GPS updates sometimes taking too much time, and not being always accurate, rather than `mapView.getController().setCenter(gp)` not being accurate. Do you agree with that?

**Comment by johan974** [comment]
Do not agree. For about 4 years I used previous versions and had no significant issues with this. Now centering the map takes often a lot of time. Sometimes it cannot find the center at all. 

Be sure - I am very thankful for using the Osmdroid! This issue - I hope you can solve it!

**Comment by monsieurtanuki** [comment]
@johan974 I must say I just found something fishy inside `MyLocationNewOverlay.setLocation`.
Do you use `MyLocationNewOverlay`?

Anyway, as a debug, you can add the following overlay to your `MapView`:
````java
mMapView.getOverlays().add(new Overlay() {
	@Override
	public void draw(Canvas c, MapView osmv, boolean shadow) {
		Log.e("DebugOverlay", "projection center:" + osmv.getProjection().getCurrentCenter());
	}
});
````
And a similar log statement each time you run `mapView.getController().setCenter(gp)` (just before):
````java
Log.e("DebugOverlay", "expected center:" + gp);
````

This way we'll be able to see:
* if `setCenter` is slow
* if the center is inaccurate

Tell me what you see in those logs.

**Comment by johan974** [comment]
Yes, I use MyLocationNewOverlay. 
I have to go in the field to replay the issues. 
The first statement, is that just creating an empty overlay, but still logging on every redraw of the screen?

My last updates (without the logging): sometimes I and a fellow app user, see that the app doesn't center at all. The map is sometimes even grey.

**Comment by johan974** [comment]
I will add the debug code. I am sure that I am tracking (following) to a target waypoint. Allow me some time to add the debug code and do the field test. Another person has the same issue with the app, tracking to a waypoint and sometimes seeing just a grey area. It takes sometimes over 10s to get a good center.

**Comment by johan974** [comment]
Yes, I have the enableFollowLoction on. Today I revisited my code that I use for almost 5 years now. It is a very, very large app for supporting geocaching in all its facets. It takes quite some time to put the debug statements in different parts. 
I hoped you found some clues in the new v6.0 osmdroid code that performs the centering more slowly or not at all. Since v6 I have issues with the centering.

**Comment by monsieurtanuki** [comment]
@johan974 I frankly don't know what's going on. I'm really confident in the setCenter feature though.
I changed some suspicious code in `MyLocationNewOverlay` that might be related to this issue.

**Comment by monsieurtanuki** [comment]
PR in #1010

**Comment by monsieurtanuki** [comment]
@johan974 Can you try with version 6.0.2?
